src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs

```diff
@@ -47,16 +47,16 @@ public bool WriteClient(CodeWriter writer, ServiceClient operationGroup)
         private void WriteOperation(CodeWriter writer, ClientMethod operation, CSharpNamespace @namespace)
         {
             //TODO: Handle multiple responses
-            var schemaResponse = operation.Response.Type;
-            CSharpType? bodyType = schemaResponse != null ? _typeFactory.CreateType(schemaResponse) : null;
+            var responseBody = operation.Response.ResponseBody;
+            CSharpType? bodyType = responseBody != null ? _typeFactory.CreateType(responseBody.Value) : null;
             CSharpType? headerModelType = operation.Response.HeaderModel != null ? _typeFactory.CreateType(operation.Response.HeaderModel.Name) : null;
 
             CSharpType responseType = bodyType switch
             {
-                null when headerModelType == null => new CSharpType(typeof(Response)),
-                null => new CSharpType(typeof(ResponseWithHeaders<>), headerModelType),
+                null when headerModelType != null => new CSharpType(typeof(ResponseWithHeaders<>), headerModelType),
                 { } when headerModelType == null => new CSharpType(typeof(Response<>), bodyType),
                 { } => new CSharpType(typeof(ResponseWithHeaders<>), bodyType, headerModelType),
+                _ => new CSharpType(typeof(Response)),
```

Comment:
Resharper was like, "You ain't covering all those cases." And I was like, "Hmm, really?" So, I just put the 'all null' case as the default.